app/alby.tsx

```diff
@@ -1,34 +1,80 @@
 import { webln } from 'alby-js-sdk'
 import { Stack } from 'expo-router'
-import { useEffect } from 'react'
-import { StyleSheet, Text, View } from 'react-native'
+import { useEffect, useRef, useState } from 'react'
+import { Button, StyleSheet, Text, TextInput, View } from 'react-native'
+
+declare global {
+  interface Window {
+    webln: any
+  }
+}
 
 export default function Alby() {
-  async function connectToAlby() {
-    // if (!window.webln) {
-    try {
-      const nostrWalletConnectUrl = 'https://arcade.chat/nwc-test' // Replace this with your own NWC url loading mechanism
-      console.log('so')
-      const nwc = new webln.NostrWebLNProvider({ nostrWalletConnectUrl })
-      console.log(nwc)
+  const nwcURL = useRef('')
+  const [address, onChangeAddress] = useState('')
+  const [message, setMessage] = useState('Zap')
+
+  async function initAlby() {
+    const nwc: any = webln.NostrWebLNProvider.withNewSecret()
+    nwcURL.current = nwc.getNostrWalletConnectUrl({
+      name: 'Arcade',
+      returnTo: document.location.toString(),
+    })
+    await nwc.initNWC()
```

Comment:
whoot, super cool! 

quck feedback:
you should pass the name here to the `initNWC` function: https://github.com/getAlby/alby-js-sdk/blob/master/README.md#initnwcname-string

```
nwcURL.current = nwc.getNostrWalletConnectUrl()
await nwc.initNWC({name: "Arcade" });
```

`getNostrWalletConnectUrl` returns the walletconnect secret URL. You don't need to pass any parameter there.

the `returnTo` is not needed with `initNWC` because initNWC opens the popup and listens for a response and the resolves and closes the popup, so no redirect is needed here.

(And one best practice is it to prompt the user on some interaction and not by default when the page is opened.)

Reply:
I've updated code based on your feedback, and it's work like charm 🔥